### PR TITLE
Implement async build creation process

### PR DESCRIPTION
**Pull request by ghengeveld**
This updates the CLI to use the new three-step build creation process: `announceBuild` ➡️ `publishBuild` ➡️ `verifyBuild`

Also this adds `mergeCommit` to the build info.

Relies on https://github.com/chromaui/chromatic/pull/6005
and https://github.com/chromaui/chromatic/pull/6162

### CODE REVIEW

**Comment by ghengeveld** [comment]
Published as `chromatic@6.6.0-alpha.0` for testing.

File: bin-src/tasks/initialize.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    { retries: 3 }
+  );
+
+  ctx.announcedBuild = announcedBuild;
+  ctx.isOnboarding =
+    announcedBuild.number === 1 || (announcedBuild.autoAcceptChanges && !autoAcceptChanges);
+
+  if (!ctx.isOnboarding && !ctx.git.parentCommits) {
+    ctx.log.warn(noAncestorBuild(ctx));
+  }
```
Comment (tmeasday) [inline bin-src/tasks/initialize.ts:74]: Can we use this to avoid the force rebuild problem?

**Review by tmeasday — APPROVED** [review]
Seems great!

File: bin-src/tasks/initialize.ts (hunk 1 above)
Comment (ghengeveld) [reply bin-src/tasks/initialize.ts:74]: https://linear.app/chromaui/issue/CH-1865/force-rebuild-during-onboarding

Comment (tmeasday) [reply bin-src/tasks/initialize.ts:74]: Oh, that all makes sense, thanks @ghengeveld!

**Comment by ghengeveld** [comment]
I'm calling this done but holding off the merge until we ship the backend changes.

**Comment by tmeasday** [comment]
@ghengeveld I tested this on a git repo with no origin setup and it errored with:

```
Field "slug" of required type "String!" was not provided.
```

**Comment by ghengeveld** [comment]
> @ghengeveld I tested this on a git repo with no origin setup and it errored with:
> 
> ```
> Field "slug" of required type "String!" was not provided.
> ```

Yeah, slug is required with the `announceBuild` mutation. If not auto-detectable, users should provide it manually (i.e. through a flag).

**Pull request merged**